### DISCUSSION TITLE
How do I add custom id to langsmith through langserve endpoint?

### DISCUSSION THREAD

**Discussion by SoulEvill** [langchain-ai/langserve#606] [Q&A]
I am serving my llm backend through langserve and logging with langsmith. It works great, however, when client side call the llm we would like to add custom id to langserver endpoint so it can log this id along with endpoint name in the langsmith. 

Are there examples of how to go about this? 

Thanks!

**Comment by eyurtsev** [comment]
Is the ID dynamic or static?

If static it's really easy:

All runnables support tags and metadata.

```python
runnable.with_config({'tags': [ 'my_tag' ], 'metadata': { 'custom_id': 'foo'})
```

If dynamic, you can achieve the same result using the with `per_request_modifier` functionality in add_routes
